BLUESPAWN-client/src/yara/args.c

```diff
@@ -0,0 +1,293 @@
+/*
+Copyright (c) 2014. The YARA Authors. All Rights Reserved.
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this
+list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation and/or
+other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its contributors
+may be used to endorse or promote products derived from this software without
+specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include <assert.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include <yara.h>
+
+#include "args.h"
+
+#define args_is_long_arg(arg)  \
+    (arg[0] == '-' && arg[1] == '-' && arg[2] != '\0')
+
+
+#define args_is_short_arg(arg)  \
+    (arg[0] == '-' && arg[1] != '-' && arg[1] != '\0')
+
+
+args_option_t* args_get_short_option(
+    args_option_t *options,
+    const char opt)
+{
+  while (options->type != ARGS_OPT_END)
+  {
+    if (opt == options->short_name)
+      return options;
+
+    options++;
+  }
+
+  return NULL;
+}
+
+
+args_option_t* args_get_long_option(
+    args_option_t *options,
+    const char* arg)
+{
+  arg += 2; // skip starting --
+
+  while (options->type != ARGS_OPT_END)
+  {
+    if (options->long_name != NULL)
+    {
+      size_t l = strlen(options->long_name);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5220377)